python_modules/dagster/dagster/core/definitions/repository.py

```diff
@@ -162,3 +175,96 @@ def solid_def_named(self, name):
                     return solid.definition
 
         check.failed('Did not find ' + name)
+
+    def get_preset(self, pipeline_name, solid_name):
+        check.str_param(pipeline_name, 'pipeline_name')
+        check.str_param(solid_name, 'solid_name')
+        return self.get_presets_for_pipeline(pipeline_name).get(solid_name)
+
+    def get_presets_for_pipeline(self, pipeline_name):
+        if self.repo_config and self.repo_config.get('pipelines'):
+            return {
+                name: PipelinePreset(
```

Comment:
this seems a bit too clever by half. It's not obvious what the shape of this is at first read. Dict[str, List[PipelinePresent]] ?